### PR TITLE
docs: improve compliance docs

### PR DESCRIPTION
**Pull request by itaysk**
## Description
Added most of the general description to the compliance page, including custom compliance docs. 
the k8s and aws pages repeated the same content, so I restructured them (note that I've removed the table with all the checks because it made it hard to read the page, if we need it it should probably be in avd). This change will also help when we add compliance in the fs command

## Related issues
- related https://github.com/aquasecurity/trivy/issues/3339

## Checklist
- [ ] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [ ] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by chen-keinan** [comment]
@itaysk there also additional changes for compliance docs in this PR#3315 , should be in sync as well

**Comment by itaysk** [comment]
yes, so if the other one get merged first, I'll rebase and update. are you otherwise ok with the changes I made?

**Comment by itaysk** [comment]
rebased and added CIS

File: docs/docs/kubernetes/cli/compliance.md [hunk 1]
```diff
@@ -29,173 +33,11 @@ trivy k8s cluster --compliance=nsa --report summary --format json
...
-| 5.5.1  | Configure Image Provenance using ImagePolicyWebhook admission controller (Manual)                      |
-| 5.7.1  | Create administrative boundaries between resources using namespaces (Manual)                           |
-| 5.7.2  | Ensure that the seccomp profile is set to docker/default in your pod definitions                       |
-| 5.7.3  | Apply Security Context to Your Pods and Containers                                                     |
-| 5.7.4  | The default namespace should not be used                                                               |
-```
-</details>
+| Compliance | Name for command | More info
+--- | --- | ---
+NSA, CISA Kubernetes Hardening Guidance v1.2 | `nsa` | [Link](https://media.defense.gov/2022/Aug/29/2003066362/-1/-1/0/CTR_KUBERNETES_HARDENING_GUIDANCE_1.2_20220829.PDF)
```
Comment (knqyf263) [inline docs/docs/kubernetes/cli/compliance.md:42]: We rename `nsa` to `k8s-nsa-1.2` or something similar, right?

**Review by knqyf263 — APPROVED** [review]
Looks nice!

File: docs/docs/kubernetes/cli/compliance.md (hunk 1 above)
Comment (knqyf263) [reply docs/docs/kubernetes/cli/compliance.md:42]: Chen told me our compliance report currently supports v1.0, not v1.2.
https://github.com/aquasecurity/trivy/pull/3349#discussion_r1058832132

**Comment by itaysk** [comment]
updated with latest changes. also note how I changed the comments in the custom report example with some more guidance

**Comment by itaysk** [comment]
did another change in latest commit, to add the general flags to the main compliance doc. now I'm not sure the aws and k8s specific docs are needed but kept them as examples. If you don't like it I can revert last commit https://github.com/aquasecurity/trivy/pull/3340/commits/158b17df6cdef1a4393a997fdbc7724db7602e6a

**Comment by knqyf263** [comment]
@itaysk Would you resolve conflicts?

**Pull request merged**